arbnode/sequencer.go

```diff
@@ -114,15 +118,87 @@ func (i *txQueueItem) returnResult(err error) {
 	close(i.resultChan)
 }
 
+type nonceCache struct {
+	cache *containers.LruCache[common.Address, uint64]
+	block common.Hash
+	dirty *types.Header
+}
+
+func newNonceCache(size int) *nonceCache {
+	return &nonceCache{
+		cache: containers.NewLruCache[common.Address, uint64](size),
+		block: common.Hash{},
+		dirty: nil,
+	}
+}
+
+func (c *nonceCache) Matches(header *types.Header) bool {
+	if c.dirty != nil {
+		return c.dirty == header
```

Comment:
I think a pointer comparison is safer because we might update parts of the header (in particular I think we update the gas used) as we build the block, but the pointer shouldn't change.